source/common/ur_params.hpp

```diff
@@ -0,0 +1,7780 @@
+        os << "UR_IMAGE_INFO_ROW_PITCH";
+        break;
+
+    case UR_IMAGE_INFO_SLICE_PITCH:
+        os << "UR_IMAGE_INFO_SLICE_PITCH";
+        break;
+
+    case UR_IMAGE_INFO_WIDTH:
+        os << "UR_IMAGE_INFO_WIDTH";
+        break;
+
+    case UR_IMAGE_INFO_HEIGHT:
+        os << "UR_IMAGE_INFO_HEIGHT";
+        break;
+
+    case UR_IMAGE_INFO_DEPTH:
+        os << "UR_IMAGE_INFO_DEPTH";
+        break;
+    default:
+        os << (value == 0 ? "0" : "invalid enumerator");
+        break;
+    }
+    return os;
+}
+inline std::ostream &operator<<(std::ostream &os,
+                                const struct ur_image_format_t params) {
+    os << "(struct ur_image_format_t){";
+
+    os << ".channelOrder = ";
+
+    os << (params.channelOrder);
+
+    os << ", ";
+    os << ".channelType = ";
+
+    os << (params.channelType);
+
+    os << "}";
+    return os;
+}
+inline std::ostream &operator<<(std::ostream &os,
+                                const struct ur_image_desc_t params) {
+    os << "(struct ur_image_desc_t){";
+
+    os << ".stype = ";
+
+    os << (params.stype);
+
+    os << ", ";
+    os << ".pNext = ";
+
+    serializeStruct(os, (params.pNext));
+
+    os << ", ";
+    os << ".type = ";
+
+    os << (params.type);
+
+    os << ", ";
+    os << ".width = ";
+
+    os << (params.width);
+
+    os << ", ";
+    os << ".height = ";
+
+    os << (params.height);
+
+    os << ", ";
+    os << ".depth = ";
+
+    os << (params.depth);
+
+    os << ", ";
+    os << ".arraySize = ";
+
+    os << (params.arraySize);
+
+    os << ", ";
+    os << ".rowPitch = ";
+
+    os << (params.rowPitch);
+
+    os << ", ";
+    os << ".slicePitch = ";
+
+    os << (params.slicePitch);
+
+    os << ", ";
+    os << ".numMipLevel = ";
+
+    os << (params.numMipLevel);
+
+    os << ", ";
+    os << ".numSamples = ";
+
+    os << (params.numSamples);
+
+    os << "}";
+    return os;
+}
+inline std::ostream &operator<<(std::ostream &os,
+                                const struct ur_buffer_properties_t params) {
+    os << "(struct ur_buffer_properties_t){";
+
+    os << ".stype = ";
+
+    os << (params.stype);
+
+    os << ", ";
+    os << ".pNext = ";
+
+    serializeStruct(os, (params.pNext));
+
+    os << ", ";
+    os << ".pHost = ";
+
+    serializePtr(os, (params.pHost));
+
+    os << "}";
+    return os;
+}
+inline std::ostream &
+operator<<(std::ostream &os,
+           const struct ur_buffer_channel_properties_t params) {
+    os << "(struct ur_buffer_channel_properties_t){";
+
+    os << ".stype = ";
+
+    os << (params.stype);
+
+    os << ", ";
+    os << ".pNext = ";
+
+    serializeStruct(os, (params.pNext));
+
+    os << ", ";
+    os << ".channel = ";
+
+    os << (params.channel);
+
+    os << "}";
+    return os;
+}
+inline std::ostream &
+operator<<(std::ostream &os,
+           const struct ur_buffer_alloc_location_properties_t params) {
+    os << "(struct ur_buffer_alloc_location_properties_t){";
+
```

Comment:
Can we get rid of those newlines (or ideally, put all `os << ...` in a single line) to make this file smaller?

Reply:
I prioritized mako template readability over cleanliness of the output file. It gets formatted afterward anyway.

I found one newline to remove from this part of the script.